gitclone/checkout_helper.go

```diff
@@ -0,0 +1,146 @@
+package gitclone
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command/git"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type fetchOptions struct {
+	// Fetch tags ("--tags")
+	tags bool
+	// Clone depth ("--depth=")
+	depth int
+}
+
+func (t fetchOptions) IsFullDepth() bool {
+	return t.depth == 0
+}
+
+const branchRefPrefix = "refs/heads/"
+
+type fetchRef struct {
+	remote, ref string
+}
+
+func newOriginFetchRef(ref string) *fetchRef {
+	return &fetchRef{
+		remote: defaultRemoteName,
+		ref:    ref,
+	}
+}
+
+func fetch(gitCmd git.Git, traits fetchOptions, ref *fetchRef) error {
+	var opts []string
+	if traits.depth != 0 {
+		opts = append(opts, "--depth="+strconv.Itoa(traits.depth))
+	}
+	if traits.tags {
+		opts = append(opts, "--tags")
+	}
+	if ref != nil {
+		opts = append(opts, ref.remote, ref.ref)
+	}
+
+	// Not neccessarily a branch, can be tag too
+	branch := ""
```

Comment:
NOTE: fetchRef is might not the best name for a struct storing a remote and a ref.
You usually name the instances of this struct `ref`, which blocks us using `ref` var name for git refs.

In this line branch variable stores a git ref, but we can not call it `ref` as it is already used for `ref *fetchRef` function argument.

Reply:
Any suggestions?

Reply:
separated parameters